src/main/cpp/src/cast_string_to_float.cu

```
@@ -0,0 +1,716 @@
+{
+  size_type const tid = threadIdx.x + (blockDim.x * blockIdx.x);
+  size_type const row = tid / 32;
+  if (row >= num_rows) { return; }
+
+  __shared__ uint64_t ipow[19];
+  if (threadIdx.x == 0) {
+    ipow[0]  = 1;
+    ipow[1]  = 10;
+    ipow[2]  = 100;
+    ipow[3]  = 1000;
+    ipow[4]  = 10000;
+    ipow[5]  = 100000;
+    ipow[6]  = 1000000;
+    ipow[7]  = 10000000;
+    ipow[8]  = 100000000;
+    ipow[9]  = 1000000000;
+    ipow[10] = 10000000000;
+    ipow[11] = 100000000000;
+    ipow[12] = 1000000000000;
+    ipow[13] = 10000000000000;
+    ipow[14] = 100000000000000;
+    ipow[15] = 1000000000000000;
+    ipow[16] = 10000000000000000;
+    ipow[17] = 100000000000000000;
+    ipow[18] = 1000000000000000000;
+  }
+  __syncthreads();
+
+  // convert
+  string_to_float<T, block_size> convert(
+    out, validity, ansi_except, valid_count, chars, offsets, ipow, incoming_null_mask, num_rows);
+  convert();
+}
+
+}  // namespace detail
+
+/**
+ * @brief Convert a string column into an float column.
+ *
+ * @param dtype Type of column to return.
+ * @param string_col Incoming string column to convert to integers.
+ * @param ansi_mode If true, strict conversion and throws on error.
+ *                  If false, null invalid entries.
+ * @param stream Stream on which to operate.
+ * @param mr Memory resource for returned column
+ * @return std::unique_ptr<column> Integer column that was created from string_col.
+ */
+std::unique_ptr<column> string_to_float(data_type dtype,
+                                        strings_column_view const& string_col,
+                                        bool ansi_mode,
+                                        rmm::cuda_stream_view stream,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  if ((string_col.size() == 0) ||
+      (dtype != data_type{type_id::FLOAT32} && dtype != data_type{type_id::FLOAT64})) {
+    return std::make_unique<column>();
+  }
```

This should throw an exception instead, because non-float target types are invalid input.